app/plugins/run-logger.js

```diff
@@ -176,32 +247,23 @@ module.exports = {
       const time = [Math.floor(req.clear_time / 1000 / 60), seconds];
       entry.time = `${time[0]}:${time[1]}`;
     }
-
-    if (reward.crate) {
-      entry.mana = reward.crate.mana ? entry.mana + reward.crate.mana : entry.mana;
-      entry.energy = reward.crate.energy ? entry.energy + reward.crate.energy : entry.energy;
-      entry.crystal = reward.crate.crystal ? entry.crystal + reward.crate.crystal : entry.crystal;
-
-      if (reward.crate.rune) {
-        const rune = reward.crate.rune;
-        entry.drop = 'Rune';
-        entry.grade = `${rune.class}*`;
-        entry.sell_value = rune.sell_value;
-        entry.set = gMapping.rune.sets[rune.set_id];
-        entry.slot = rune.slot_no;
-        entry.efficiency = gMapping.getRuneEfficiency(rune).current;
-        entry.rarity = gMapping.rune.class[rune.sec_eff.length];
-        entry.main_stat = gMapping.getRuneEffect(rune.pri_eff);
-        entry.prefix_stat = gMapping.getRuneEffect(rune.prefix_eff);
-
-        rune.sec_eff.forEach((substat, i) => {
-          entry[`sub${i + 1}`] = gMapping.getRuneEffect(substat);
-        });
-      } else {
-        entry.drop = this.getItem(reward.crate);
+    if (command === 'BattleDungeonResult_V2') {
+      entry.mana = reward.mana ? entry.mana + reward.mana : entry.mana;
+      entry.energy = reward.energy ? entry.energy + reward.energy : entry.energy;
+      entry.crystal = reward.crystal ? entry.crystal + reward.crystal : entry.crystal;
+    } else {
+      if (reward.crate) {
+        entry.mana = reward.crate.mana ? entry.mana + reward.crate.mana : entry.mana;
+        entry.energy = reward.crate.energy ? entry.energy + reward.crate.energy : entry.energy;
+        entry.crystal = reward.crate.crystal ? entry.crystal + reward.crate.crystal : entry.crystal;
       }
     }
 
+    entry =
```

Comment:
You either need to check whether it's a win or a loss here or add checks in getItemV2 whether the 1st item exists.

When the run errors, it is not logged as view dos not exist on undefined (since you pass empty array the first item is always undefined).